syft/file/digest.go

```diff
@@ -1,6 +1,62 @@
+// nolint:gosec
 package file
 
+import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"io"
+	"os"
+
+	"github.com/anchore/syft/internal/log"
+)
+
+type HashAlgorithm string
+
+const (
+	HashAlgoMD5    HashAlgorithm = "MD5"
+	HashAlgoSHA1   HashAlgorithm = "SHA-1"
+	HashAlgoSHA256 HashAlgorithm = "SHA-256"
+)
+
 type Digest struct {
 	Algorithm string `json:"algorithm"`
 	Value     string `json:"value"`
 }
+
+func CalculateDigest(filepath string, algorithm HashAlgorithm) (digest *Digest, err error) {
```

Comment:
this logic already exists in the digests cataloger and is capable of handing multiple digests at once cheaply. I think we could extract that out and use it here. 

https://github.com/anchore/syft/blob/v0.43.2/syft/file/digest_cataloger.go#L80-L118